### PR TITLE
Cleaned up and added some new information

### PR DESCRIPTION
**Pull request by taataam**
I cleaned up the advection non-uniform solver and removed the redundant test problems, keeping only the slotted problem. Also added the required solver's parameters the solver's inc file. I added my information to the paper as well.

### CODE REVIEW

**Comment by taataam** [comment]
The fail is due to flake8. It gives the [F405](https://lintlyci.github.io/Flake8Rules/rules/F405.html) error for all the `__init__.py` files.

File: advection_nonuniform/advective_fluxes.py [hunk 1]
```diff
@@ -110,17 +110,17 @@ def unsplit_fluxes(my_data, rp, dt, scalar_name):
...
     for index, vel in np.ndenumerate(v.v(buf=1)):
-        F_y.v(buf=1)[index] = vel * (a_y.v(buf=1)[index] -
-                                     dtdx2 * (F_xt.ip_jp(1, shift_y.v(buf=1)[index], buf=1)[index] -
-                                              F_xt.jp(shift_y.v(buf=1)[index], buf=1)[index]))
+        F_y.v(buf=1)[index] = vel * (a_y.v(buf=1)[index] - dtdx2 *
+               (F_xt.ip_jp(1, shift_y.v(buf=1)[index], buf=1)[index] -
+                F_xt.jp(shift_y.v(buf=1)[index], buf=1)[index]))
 
-    return F_x, F_y
+    return F_x, F_y
```
Comment (harpolea) [inline advection_nonuniform/advective_fluxes.py]: This deletion of the new line is causing `flake8` to complain

**Comment by harpolea** [comment]
This looks like it's because `flake8` has been upgraded again - with 3.6 it was fine, but when I run with 3.7.5 I now see F405 errors. I'll fix these on master.

I also see the error 

`/advection_nonuniform/advective_fluxes.py:126:20: W292 no newline at end of file`

- this is pretty trivial, but would you mind fixing that? Otherwise the changes look good to me!

**Comment by harpolea** [comment]
Ok, I've fixed all the flake8 F405 complaints now. If you could just fix that other tiny flake8 issue then this should be good to merge

**Comment by taataam** [comment]
@harpolea I fixed the newline problem as well as `__init__.py` the same way you did. Just one thing, with `flake8` version 3.7.5, when I add the newline it gives error `W391` meaning that I should remove the blank line at the end on the line! I think they changed things in the new versions and no blank line should be at the end of the file.

**Comment by harpolea** [comment]
Hmm ok, I think this annoyingly may be something to do with the text editor you're using - sometimes editors add extra newline characters at the ends of files without you noticing. If I view the file on github, there's no extra line, but if I open the raw file in my text editor then there's an extra blank line there. Everything else looks good though so I'll merge then fix that file so flake8 is happy. Thanks for your quick changes!

**Comment by taataam** [comment]
@harpolea Great! I use vim for editing the files. Thank you.

**Pull request merged**